### DISCUSSION TITLE
How to set needle position without active force?

### DISCUSSION THREAD

**Discussion by linhongbin** [surgical-robotics-ai/surgical_robotics_challenge#28] [Q&A]
Hi adnan,

I write a script to reset the needle to original position:
```
from ambf_client import Client
ambf_client = Client('xxx')
ambf_client.connect()
pos = [-0.20697696629671555, 0.5583796718976854, 0.7247253773270505]
rpy = [0.030316076843698688, 0.029994417468907717, 0.036854178797618514]
needle = ambf_client.get_obj_handle('Needle')
needle.set_pos(*pos)
needle.set_rpy(*rpy)
```

But I found it hover in the air, which seems like applying force to pin the needle in the specified position.
How could disable the force or is there any other method to reset the needle by python code?

**Comment by linhongbin** [comment]
I found a solution like this:

Adding zero force to the object after setting position:
```sh
time.sleep(1) # wait for needle to move to position
needle.set_force(0, 0, 0)
needle.set_torque(0, 0, 0)
```

The needle will be moved to the desired position. But the position of the thread seems to be random. But this solution is good to me for now.

**Comment by adnanmunawar** [comment]
Hello Hongbin,
As you already noticed, calling `set_pos()` / `set_rpy()` will keep the pos controller active, and thus calling the `set_force(0,0,0)` / `set_torque(0,0,0)` is a recommended way to disable the pos controller to free up the needle. To reset all the rigid bodies (including the thread), you may use the keyboard shortcut 'CTRL + R'. 

Alternatively, we can implement a way to reset the simulation using the scripts if that is something that you are interested in.

**Reply by linhongbin** [reply]
Thanks for your reply. How to reset the simulation using script? It would be of great help to ensure the reproducibility.

**Comment by adnanmunawar** [comment]
Hi Hongbin,
Please check out this [PR](https://github.com/WPI-AIM/ambf/pull/160). You'll need to update and rebuild AMBF for this functionality. The instructions on using the reset features are mentioned in the PR [description](https://github.com/WPI-AIM/ambf/pull/160#issue-1174865205).

**Reply by linhongbin** [reply]
Great, looking forward to using  the new feature!
